Source/DafnyLanguageServer/Language/Symbols/DafnyLangSymbolResolver.cs

```diff
@@ -150,9 +154,60 @@ private class SymbolDeclarationResolver {
           cancellationToken.ThrowIfCancellationRequested();
           functionSymbol.Parameters.Add(ProcessFormal(scope, parameter));
         }
+        if (function.Result != null) {
+          functionSymbol.Parameters.Add(ProcessFormal(scope, function.Result));
+        }
+
+        Func<Expression, ScopeSymbol?> subP = (Expression e) => ProcessFunctionExpr(functionSymbol, e);
+        functionSymbol.Body = ProcessFunctionExpr(functionSymbol, function.Body);
+        functionSymbol.Ens = ProcessListAttributedExpressions(function.Ens, subP);
+        functionSymbol.Req = ProcessListAttributedExpressions(function.Req, subP);
+        functionSymbol.Reads = ProcessListFramedExpressions(function.Reads, subP);
+        functionSymbol.Decreases = ProcessListExpressions(function.Decreases.Expressions, subP);
+        functionSymbol.ByMethodBody = ProcessFunctionByMethodBody(functionSymbol, function.ByMethodBody);
         return functionSymbol;
       }
 
+      private List<ScopeSymbol> ProcessListAttributedExpressions(
+        IList<AttributedExpression> list, Func<Expression, ScopeSymbol?> subProcess) {
+        return list.SelectMany(atExpr => Symbol.AsEnumerable<ScopeSymbol>(
+          subProcess(atExpr.E))).ToList();
+      }
```

Comment:
`SelectMany` appears to be unneccessary to me. Why not like this?
```cs
        return list.Select(atExpr => subProcess(atExpr.E)).ToList();
```

Reply:
We don't want nulls in this list, and Select keeps those.